homeassistant/components/mycroft/notify.py

```diff
@@ -24,16 +25,17 @@ def get_service(
 class MycroftNotificationService(BaseNotificationService):
     """The Mycroft Notification Service."""
 
-    def __init__(self, mycroft_ip):
+    def __init__(self, mycroft_ip) -> None:
         """Initialize the service."""
         self.mycroft_ip = mycroft_ip
+        self.bus = MessageBusClient(host=self.mycroft_ip)
+        self.bus.run_in_thread()
```

Comment:
We need to clean up the thread when Home Assistant stops. We can listen for the Home Assistant stop event and register an event handler that closes the websocket.

Alternatively, since we don't actually need to listen for incoming websocket messages, maybe just use this helper function that doesn't keep the websocket open?

https://github.com/MycroftAI/mycroft-messagebus-client/blob/01ade972facb9c2287d4b9eade49b7f8c1e74655/mycroft_bus_client/send_func.py#L30

Reply:
Thank you for your educational feedback. As that package still depends on the pinned pyee and the development of mycroft has stopped, I decided to fork the mycroft-messagebus-client and resolve the issue myself. Having the websocket open on load gives an early warning about potential issues before it's time to deliver the message, so I'd prefer using the websocket as intended. Plus I'd get some experience publishing on PyPi. Implementing this using plain websocket-client would be fairly simple, but that would then lead to having to pin websocket-client and I'm pretty sure that could lead to much collateral damage and enemies I don't want to make.